bcl2fastq/lib/bcl2fastq_utils.py

```diff
@@ -297,6 +297,15 @@ def construct_command(self):
         """
         raise NotImplementedError("Subclasses should implement this!")
 
+    def delete_output(self):
+        """
+        Delete the output directory if it exists
+        :return: None
+        """
+        if os.path.isdir(self.config.output):
+            log.info("Found a directory at output path {}, will remove it.".format(self.config.output))
+            shutil.rmtree(self.config.output)
```

Comment:
I assume that the rights of the user running the service will limit risk to system directories (i.e. that it's never running as a privileged user).

But if you want to be extra certain that you're not deleting something accidentally, you could add a guard where you only allow a certain naming standard for the output directories (e.g. requiring that the name starts with `arteria_output_` or that it's a (grand)child of some directory.) I would suggest applying such a guard here, but not when reading the config, to limit the risk of refactoring removing the condition accidentally.

I'm a fan of such guards, since following a naming convention wouldn't limit the usability of the application (I assume) but it would be very unlikely to break.